### ISSUE TITLE
Images with Tags I.e. not the "latest" are updated frequently

### ISSUE THREAD

**Issue by Nanoemc** [CoffeaTeam/af-images#35]
Some of the 0.7 images (I.e. coffea-base-almalinux8
) and coffea 2024 images (I.e. coffea-dask-almalinux8) have their package versions updated at the same frequency as the latest images. It causes issues with analysis pipelines and with the EAF, maybe there should be a latest-2024-py3.x tag and a latest-0.7.22-py3.x tag to avoid this in the future. It is hard to have a stable pipeline when some of the tagged images (not latest) are changing.

**Comment by nsmith-** [comment]
So, concretely, we do have what I would call a bug in the github action: when we bump a version of coffea-dask (i.e. coffea CalVer) we re-render also the coffea-base image, which may cause some unpinned versions to be updated and potentially break if the tests do not catch issues. For example, when #33 was merged the CI pushed a new image overwriting the tag `coffea-base-almalinux8:0.7.22-py3.10`: https://github.com/CoffeaTeam/af-images/actions/runs/11013386721/job/30581631518
We probably want all tags but `latest` to be permanent.

**Comment by nsmith-** [comment]
Perhaps we can open a new issue but there is more to discuss with respect to image version tags. For example, we may want to start a separate af-image versioning scheme rather than staying in lock-step with coffea.

**Comment by oshadura** [comment]
oh, I didnt want to close it, I just want to test how it work now the worksflow

**Comment by oshadura** [comment]
Now I think I have the workflow similiar to what is doing dask:

- we generate repository tags each time there is a new version of coffea released and when tag is pushed we build only once the image [2024.9.1-py3.11](https://hub.docker.com/layers/coffeateam/coffea-dask-almalinux9/2024.9.1-py3.11/images/sha256-0f96757378fca5019dbfa26ad9f26ea97bdbd89108d5478c7322f1560a903b5d?context=explore) == permanent tag
- each time we merge something in the `main` branch we push the [dev-py3.10](https://hub.docker.com/layers/coffeateam/coffea-dask-almalinux9/dev-py3.10/images/sha256-4ac1e56fc817ac1095d23005c480b58440499de27158a5519d1252a1fa0f4295?context=explore) tag == not permanent
- probably latest, latest-pyxxx then could gone [latest-py3.11](https://hub.docker.com/layers/coffeateam/coffea-dask-almalinux9/latest-py3.11/images/sha256-0f96757378fca5019dbfa26ad9f26ea97bdbd89108d5478c7322f1560a903b5d?context=explore)
